Airside_Functions.cpp

```diff
@@ -0,0 +1,458 @@
+/**************************************************************************************************/
+// Author: Jingting Liu
+// April 9th, 2021
+
+// to run the test without cmake:
+// gcc -g Airside_Functions.cpp
+// a.out
+
+// to run the test with cmake:
+// mkdir build, cd build
+// cmake ..
+// make
+// execute generated target
+
+
+// refer to this page for the stucture of mavlink messages 
+// https://mavlink.io/en/guide/serialization.html
+
+// The decoder and encoder only support GPS and gimbal control, other simpler commands will be taken
+// care of by the Xbee communication directly
+/**************************************************************************************************/
+
+#include "Mavlink2_lib/common/mavlink.h"
+#include "Airside_Functions.hpp"
+
+mavlink_decoding_status_t Mavlink_airside_decoder(PIGO_Message_IDs_e* type, uint8_t incomingByte, uint8_t *telemetryData)
+{
+    int channel = MAVLINK_COMM_0; //mavlink default one channel
+    PIGO_Message_IDs_e decoded_message_type = MESSAGE_ID_NONE;
+    mavlink_decoding_status_t decoding_status = MAVLINK_DECODING_INCOMPLETE;
+
+    mavlink_status_t status;
+    memset(&status, 0x00, sizeof(mavlink_status_t));
+
+    mavlink_message_t decoded_msg;
+    memset(&decoded_msg, 0, sizeof(mavlink_message_t));
+
+    uint8_t message_received = mavlink_parse_char(channel, incomingByte, &decoded_msg, &status);
+
+    if (message_received)
+    {
+        if (telemetryData == NULL)
+        {
+            return MAVLINK_DECODING_FAIL;
+        }
+
+        switch(decoded_msg.msgid)
+        {
+            case MAVLINK_MSG_ID_GLOBAL_POSITION_INT: // ID for GLOBAL_POSITION_INT, 33
+                {
+                    mavlink_global_position_int_t global_position;
+                    memset(&global_position, 0x00, sizeof(mavlink_global_position_int_t));
+
+                    mavlink_msg_global_position_int_decode(&decoded_msg, &global_position);
+                    uint32_t warg_ID = global_position.time_boot_ms;
+
+                        if (warg_ID == MESSAGE_ID_GPS_LANDING_SPOT)
```

Comment:
Maybe put a comment explaining what is going on here. I remember vaguely that we decided to use the gps struct as a template so we can cheat our own structs in, but I am not 100% sure. A comment would be great tho! 

But damn, this is really cool stuff!